pkg/mcp/server/server.go

```diff
@@ -298,6 +299,9 @@ func New(watcher Watcher, supportedTypes []string, authChecker AuthChecker, repo
 }
 
 func (s *Server) newConnection(stream mcp.AggregatedMeshConfigService_StreamAggregatedResourcesServer) (*connection, error) {
+	if err := s.newConnectionLimiter.Wait(stream.Context()); err != nil {
```

Comment:
Can you add a test for connection rate limiting if you add this line?

Reply:
Sure, when i am free.